### PR TITLE
Update readme, fix the text format, custom the message type

### PR DESCRIPTION
**Pull request by YuhanHe**
Update readme, fix the text format, custom the message type

### CODE REVIEW

**Review by daleront — APPROVED** [review]

File: lib/commit_summary.bash [hunk 1]
```diff
@@ -7,6 +7,7 @@ function commit_summary() {
     local TO="$2"
 
     # print out the commits that have been added
+    git pull --tags
```
Comment (RodionovDmitry) [inline lib/commit_summary.bash:10]: For reviewers: We need this if we want to use tag names in FROM and TO below. Looks like tags are not always pulled with the code on the agents.

**Review by RodionovDmitry — APPROVED** [review]

File: hooks/post-checkout [hunk 2]
```diff
@@ -31,16 +34,16 @@ if [[ "${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_COMMIT_SUMMARY_ENABLED:-false}" =~ (tr
...
+    MESSAGE=$(echo "$MESSAGE" | sed 's/"/\\"/g')
 fi
 
 CURRENT_COMMIT=$(git rev-parse HEAD)
 THREAD_KEY=${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_THREAD_KEY:=$CURRENT_COMMIT}
 
 FAIL=0
 ROWS_JSON="none"
-if [ $TYPE == 'CARD' ]
+if [ $TYPE == 'KeyValueCARD' ]
```
Comment (JonCubed) [inline hooks/post-checkout:52]: what is the reason for changing to `KeyValueCARD`? It's a breaking change

Comment (YuhanHe) [reply hooks/post-checkout:52]: Hi Jonathon, the type of the card message we used is key-value one. Just want to make the name of the variable more meaningful.

Comment (VanyaChe) [reply hooks/post-checkout:52]: I agree with @YuhanHe  that CARD is a too general term to describe a message type that is used here. As card by itself has many [subtypes](https://developers.google.com/hangouts/chat/reference/message-formats/cards), the distinction should be made between them. Using gchat card types seem sensible step here.

Though, to keep backwards compatibility I suggest adding normalisation of parameter at the beginning of the script. Something like below should do the work:

```bash
if [ $TYPE == 'CARD' ]
then
    TYPE='KeyValueCARD'
fi
````

The documentation needs to be modified to say that `CARD` value  of `type` parameter is obsolete and will be dropped in the future

Comment (JonCubed) [reply hooks/post-checkout:52]: I'm fine with your proposal @VanyaChe

File: hooks/post-checkout [hunk 3]
```diff
@@ -11,13 +11,17 @@ echo "--- Sending Hangouts Message"
...
-WEBHOOK_URL2="${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_WEBHOOK_URL2?}"
-WEBHOOK_KEY="${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_WEBHOOK_KEY?}"
-USERTOKEN_FAIL="${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_USERTOKEN_FAIL?}"
-USERTOKEN_PASS="${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_USERTOKEN_PASS?}"
 TYPE=${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_TYPE}
-CARD_DETAILS=${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_CARD_DETAILS}
-ROWS=$(echo "${BUILDKITE_PLUGIN_HANGOUTS_NOTIFY_ROWS?}" | envsubst)
+if [ $TYPE == 'CARD' ]
+  then
+  TYPE="KeyValueCARD"
```
Comment (VanyaChe) [inline hooks/post-checkout:18]: 1. I don't think resetting type value here makes much sense as it means that plugin clients still have to pass `CARD` value for `TYPE` parameter ( given this `if [ $TYPE == 'CARD' ]`)
 
    -  I wanted us to keep backward compatibility by having a **separate** `if` statement that normalises TYPE to a supported value `KeyValueCARD` like that:
    ```bash
    if [ $TYPE == 'CARD' ]
      then
      TYPE="KeyValueCARD"
    fi
    ```

     - After TYPE was normalised to be `KeyValueCard` you can use it in `if` statement on line 15. 

    ```bash
    if [ $TYPE == 'KeyValueCARD' ]
      then
      # some code here
    fi
    ```
2. Could you update the documentation to say that `CARD` is redundant and won't be supported in future, plus `KeyValueCard `should be used instead?

**Review by VanyaChe — APPROVED** [review]

**Comment by VanyaChe** [comment]
@YuhanHe let's close this PR?

**Pull request closed without merging**